src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/serializer/IdentitySet.kt

```diff
@@ -0,0 +1,29 @@
+package it.krzeminski.snakeyaml.engine.kmp.serializer
+
+import it.krzeminski.snakeyaml.engine.kmp.internal.IdentityHashCode
+import it.krzeminski.snakeyaml.engine.kmp.internal.identityHashCode
+
+/**
+ * A set that compares objects by their identities, not values.
+ * It's an attempt to reimplement `Collections.newSetFromMap(new IdentityHashMap<Node, Boolean>())`
+ * from the JVM.
```

Comment:
suggestion: I would also add a note that this class won't work if the `T` is a primitive type or a string

Reply:
Sure, will add.

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/272/commits/0c122e04fd2a8a14f1c898de13251774b169ba7d